corehq/apps/geospatial/static/geospatial/js/models.js

```diff
@@ -808,7 +809,7 @@ hqDefine('geospatial/js/models', [
                         );
                         // redraw using mapControlsModelInstance
                         self.selectedSavedPolygonId(ret.id);
-                        self.shouldRefreshPage(true);
+                        self.shouldRefreshPage(self.requiresPageRefresh());
```

Comment:
This is one of the two scenarios (the other is delete area) where refresh is required because saved polygons filter populates with page load. I know for delete area we went with auto refresh and for [new area](https://github.com/dimagi/commcare-hq/pull/34842) we went with the banner approach. 
We could probably keep this as it is.

Reply:
Good point regarding creating a new saved polygon. If the user were to change case filters then the newly saved polygon would disappear from the list until a refresh is done. I'll revert this so that a refresh request is still made when a polygon is saved.

Reply:
Addressed in [0b515bf](https://github.com/dimagi/commcare-hq/pull/35300/commits/0b515bfc840b8461e5e24110be3d43ccb954bda6)

Reply:
@zandre-eng Looks like this [0b515bf](https://github.com/dimagi/commcare-hq/pull/35300/commits/0b515bfc840b8461e5e24110be3d43ccb954bda6) is an accidental commit . The mobile ucr script got included :)

Reply:
Ahhh, my mistake. Yeah I accidentally committed the wrong file, however I've done a rebase and the right one is committed now [a13bcdf](https://github.com/dimagi/commcare-hq/pull/35300/commits/a13bcdf6f60a0e1e7f5da3931245845d5ab9d39a)

Reply:
No worries